packages/framer-motion/src/render/utils/__tests__/animation-state.test.ts

```diff
@@ -1,47 +1,59 @@
-import { ResolvedValues, VisualElement } from "../../types"
+import { ResolvedValues } from "../../types"
 import { AnimationState, createAnimationState } from "../animation-state"
 import { AnimationType } from "../types"
 import { checkTargetForNewValues, getOrigin } from "../setters"
-import { visualElement } from "../../"
 import { MotionProps } from "../../../motion/types"
 import { createHtmlRenderState } from "../../html/utils/create-render-state"
 import { createBox } from "../../../projection/geometry/models"
+import { VisualElement } from "../../VisualElement"
+import { TargetAndTransition } from "../../.."
 
-const stateVisualElement = visualElement<
+class StateVisualElement extends VisualElement<
```

Comment:
Is this one only for testing? Should we move it to it's own file?

Reply:
Done!